Assets/PlayerMovement.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour
+{
+
+    public CharacterController Controller;
+
+    [SerializeField]
+    public float MovementSpeed = 2f;
+    public float Speed = 12f;
+    public float Gravity = -9.81f;
+    public float JumpHeight = 3f;
+
+    public Transform GroundCheck;
+    public float GroundDistance = 4f;
+    public LayerMask GroundMask;
+
+    Vector3 velocity;
+    bool isGrounded;
```

Comment:
```suggestion
    private bool _isGrounded;
```

Reply:
and update usages for these below

Reply:
Do we need to use this externally? If not we don't need to store it as a field global to this class and it can be used in a method like below

Reply:
added isGrounded as a method in Update